### DISCUSSION TITLE
Enhancement: Correctly handle messages with role other than `"assistant"` in conversation history.

### DISCUSSION THREAD

**Discussion by bendavis78** [danny-avila/LibreChat#4898] [Troubleshooting]
I'm working on a custom endpoint that mimics OpenAI's completions api. This endpoint uses function calling, so any messages that include the function call results will have `role="tool"`. I want to maintain these function result messages in the conversation state (ie the `messages` array sent to server on each request) so that the function call information is kept in context.

Consider the following message history:
```
{
  "messages": [
    {
      "role": "system",
      "content": "You are an assistant that helps the user decide what to wear."
    },
    {
      "role": "user",
      "content": "Should I wear a jacket today?"
    },
    {
      "role": "assistant",
      "content": null,
      "tool_calls": [
        {
          "id": "call_YOA5bPK1g8fOi0nphUO2lw1D",
          "type": "function",
          "function": {
            "arguments": "",
            "name": "get_weather"
          }
        }
      ]
    },
    {
      "role": "tool",
      "content": "weather: sunny, temp: 10",
      "tool_call_id": "call_1234ABDEFGH"
    },
    {
      "role": "assistant",
      "content": "Yes it's a bit cold, you should wear a jacket.",
      "tool_calls": []
    }
  ]
}
```

Currently when LibreChat encounters the `role="tool"` message from the server, it assumes it is a `role="assistant"` message, and concatenates it with any following messages. The message state sent back to the server will combine the last two messages into a single system message. This loses conversation context.

It would be useful if the UI should only show messages if the `role` is set to "assistant". All other messages should be stored in the conversation history with its original attributes retained. This would allow the client to send the full history to the server and keep everything in context.

One might argue that you don't need to retain the tool response in the context, but I think that decision should be on the server side.

### Components:

General, UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
these would be retained when using Agents, will link to their docs when they are published momentarily.
